awpy/visualization/plot.py

```diff
@@ -326,3 +341,382 @@ def plot_nades(
+
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+                    If player_dict not of expected
+                    extract_player_positions output format
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
+        raise ValueError(msg)
+
+    ct_tile_set, t_tile_set = set(ct_tiles.keys()), set(t_tiles.keys())
+    all_tiles = ct_tile_set.union(t_tile_set)
+
+    # Iterate through the tiles that have a value
+    for tile in all_tiles:
+        if tile in NAV[map_name]:
+            area = NAV[map_name][tile]
+            se_x = position_transform(map_name, area["southEastX"], "x")
+            nw_x = position_transform(map_name, area["northWestX"], "x")
+            se_y = position_transform(map_name, area["southEastY"], "y")
+            nw_y = position_transform(map_name, area["northWestY"], "y")
+            width = se_x - nw_x
+            height = nw_y - se_y
+
+            # Use max value (if exists) for each side for the current tile
+            ct_val = max(ct_tiles[tile] + [0])
+            t_val = max(t_tiles[tile] + [0])
+
+            # Use each side's value above as weights for weighted average
+            # to find correct color
+            cur_color = min(ct_val / (ct_val + t_val), ct_val) * np.array(
+                [0, 1, 0]
+            ) + min(t_val / (ct_val + t_val), t_val) * np.array([1, 0, 0])
+
+            rect = patches.Rectangle(
+                (nw_x, se_y),
+                width,
+                height,
+                linewidth=1,
+                edgecolor=cur_color,
+                facecolor=cur_color,
+                alpha=1.0,
+            )
+            axes.add_patch(rect)
+        else:
+            print("Tile not found in map:", tile)
+
+    # Plot player positions if given
+    if player_data is not None:
+        t_positions, ct_positions = (
+            player_data["t"]["player-locations"],
+            player_data["ct"]["player-locations"],
+        )
+        if len(t_positions) > 0:
+            transformed_x = [
+                position_transform(map_name, loc[0], "x") for loc in t_positions
+            ]
+            transformed_y = [
+                position_transform(map_name, loc[1], "y") for loc in t_positions
+            ]
+            color_arr = ["#5d79ae"] * len(t_positions)
+            axes.scatter(transformed_x, transformed_y, c=color_arr)
+        if len(ct_positions) > 0:
+            transformed_x = [
+                position_transform(map_name, loc[0], "x") for loc in ct_positions
+            ]
+            transformed_y = [
+                position_transform(map_name, loc[1], "y") for loc in ct_positions
+            ]
+            color_arr = ["#de9b35"] * len(ct_positions)
+            axes.scatter(transformed_x, transformed_y, c=color_arr)
+    axes.axis("off")
+
+
+def plot_map_control_snapshot(
+    map_name: str,
+    tile_values: FrameMapControl,
+    player_pos: dict | None = None,
```

Comment:
Is this intended to be called independently by a user?

Reply:
Yeah I was intending to have it be a public function that users can use. If the users just have the FrameMapControl object but not the actual GameFrame object, then this function can be used for plotting.